src/client/datascience/variablesView/variableView.ts

```diff
@@ -78,29 +113,62 @@ export class VariableView extends WebviewViewHost<IVariableViewPanelMapping> imp
+                    request.variable,
+                    this.notebookEditorProvider.activeEditor.notebook
+                );
+                const title: string = `${localize.DataScience.dataExplorerTitle()} - ${request.variable.name}`;
+                await this.dataViewerFactory.create(jupyterVariableDataProvider, title);
+            }
+        } catch (e) {
+            traceError(e);
+            this.appShell.showErrorMessage(e.toString());
+        }
+    }
+
+    // Variables for the current active editor are being requested, check that we have a valid active editor
+    // and use the variables interface to fetch them and pass them to the variable view UI
     private async requestVariables(args: IJupyterVariablesRequest): Promise<void> {
-        // For now, this just returns a fake variable that we can display in the UI
-        const response: IJupyterVariablesResponse = {
-            totalCount: 1,
-            pageResponse: [
-                {
-                    name: 'test',
-                    value: 'testing',
-                    executionCount: args?.executionCount,
-                    supportsDataExplorer: false,
-                    type: 'string',
-                    size: 1,
-                    shape: '(1, 1)',
-                    count: 1,
-                    truncated: false
-                }
-            ],
-            pageStartIndex: args?.startIndex,
-            executionCount: args?.executionCount,
-            refreshCount: args?.refreshCount || 0
-        };
-
-        this.postMessage(InteractiveWindowMessages.GetVariablesResponse, response).ignoreErrors();
+        if (this.notebookEditorProvider.activeEditor && this.notebookEditorProvider.activeEditor.notebook) {
+            const response = await this.variables.getVariables(args, this.notebookEditorProvider.activeEditor.notebook);
+
+            this.postMessage(InteractiveWindowMessages.GetVariablesResponse, response).ignoreErrors();
+        }
+    }
+
+    // When the kernel state is changed we need to see if it's a cell from the active document that finished execution
+    // If so update the execution count on the variable view to refresh variables
+    private async kernelStateChanged(kernelStateEvent: KernelStateEventArgs) {
+        // Check for non-silent executes from the current cell that have an execution order
+        if (
+            kernelStateEvent.state === KernelState.executed &&
+            kernelStateEvent.cell &&
+            kernelStateEvent.cell.metadata.executionOrder &&
+            kernelStateEvent.silent !== true
+        ) {
+            // We only want to update the variable view execution count when it's the active document executing
+            if (
+                this.notebookEditorProvider.activeEditor &&
+                this.notebookEditorProvider.activeEditor.file.toString() === kernelStateEvent.resource.toString()
```

Comment:
Should you maybe do a arePathsSame here? File casing might not always be consistent.

Reply:
Doesn't this mean this new variable viewer is hardcoded to only work with Native notebooks?
I'm ok with that, just feels like we're writing code that will be difficult to change later due to this tight coupling. I.e. I'd prefer to such coupling be outside this class, ie. reduce the coupling (**no need to change, just a thought**)

Reply:
Good call I'll change it. Interestingly enough it looks like vscode has code to directly compare URIs but it doesn't look publicly accessible to me. Seems like a URI should be something that you would be able to compare without worrying about the concept of paths.

Reply:
@DonJayamanne it is currently specced for native only. Lemme make a note on this for the next PR, might be good to remove that knowledge as I think that it would be nice to support the old webviews with the view based variable explorer as well.